readthedocs/proxito/middleware.py

```diff
@@ -278,6 +279,32 @@ def add_hosting_integrations_headers(self, request, response):
             if addons:
                 response["X-RTD-Hosting-Integrations"] = "true"
 
+    def add_cors_headers(self, request, response):
+        """
+        Add CORS headers only on PUBLIC versions.
+
+        DocDiff addons requires making a request from
+        ``RTD_EXTERNAL_VERSION_DOMAIN`` to ``PUBLIC_DOMAIN`` to be able to
+        compare both DOMs and show the visual differences.
+
+        This request needs ``Access-Control-Allow-Origin`` HTTP headers to be
+        accepted by browsers. However, we cannot expose these headers for
+        documentation that's not PUBLIC.
+        """
+        project_slug = getattr(request, "path_project_slug", "")
+        version_slug = getattr(request, "path_version_slug", "")
+
+        if project_slug and version_slug:
+            allow_cors = Version.objects.filter(
+                project__slug=project_slug,
+                slug=version_slug,
+                privacy_level=PUBLIC,
+            ).exists()
+            if allow_cors:
+                response.headers["Access-Control-Allow-Origin"] = "*"
```

Comment:
Instead of using `"*"` here we could have some logic here to only support `*.readthedocs.build`

> Limiting the possible Access-Control-Allow-Origin values to a set of allowed origins requires code on the server side to check the value of the [Origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) request header, compare that to a list of allowed origins, and then if the [Origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) value is in the list, set the Access-Control-Allow-Origin value to the same value as the [Origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) value.

_(from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin)_

Reply:
I don't think we _require_ this since these versions are all public, but we can implement it in a following iteration if we consider.

Reply:
If we are going to enable this on .com, I'd prefer if we start the most restricted we can.

Reply:
Agreed -- we definitely want to err on the side of caution here given the security considerations that have kept this disabled in the past.